### PR TITLE
De ordinals

### PR DESCRIPTION
**Pull request by Bantolomeus**
I appreciate a review of the code. It feels not right to write down all 32 cases as enum, but german ordinals are partly irregular up to 19. I could not come up with a better solution.

### CODE REVIEW

**Comment by Bantolomeus** [comment]
This PR should solve the german part of https://github.com/korlibs/korge/issues/669.

**Comment by soywiz** [comment]
@Bantolomeus thanks for the PR!

Looking at the code I would suggest a change in how it is done. Let me elaborate on that:
* By using the enum, it is needed to replace some characters to support in `getOrdinalInGerman` the `ä`, `ö`, `ü`. The enum adds an overhead
* The code ends using `GermanOrdinal.values()` which generates a new array and then iterate over all the elements.
* The enum elements are not used as literals anywhere, so I would suggest a Map here. To avoid creating a new class, with all the elements, that is normally more heavyweight in some cases.

So my proposal is to do something like:

```
companion object {
  private val NUM_TO_ORD: Array<String> = arrayOf(
      "", // index 0 (invalid)
      // ...
      "zwölfte", // index 12 (no need to do the replace `oe` to `ö`)
      // ...
  )
  private val ORD_TO_NUM: Map<String, Int> = ORD_TO_NUM.mapIndexed { index, s -> s to index }.toMap()
}
```

Then accessing the array can be done with `ORD_TO_NUM.getOrNull(index) ?: error("Invalid numeral")` and the opposite with `ORD_TO_NUM[key.toLowerCase()]`.

**Comment by Bantolomeus** [comment]
@soywiz I hope deleting `companion object : GermanKlockLocale()` from the class does not break anything elsewhere, but I was not allowed to define two companion objects in the class.
Thanks for the earlier elaboration.

File: klock/src/commonMain/kotlin/korlibs/time/locale/de.kt [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
 import korlibs.time.KlockLocale
+import korlibs.time.KlockLocaleContext
+import korlibs.time.DayOfWeek
 
 val KlockLocale.Companion.german get() = GermanKlockLocale
 
 open class GermanKlockLocale : KlockLocale() {
-	companion object : GermanKlockLocale()
+
+    companion object {
```
Comment (soywiz) [inline klock/src/commonMain/kotlin/korlibs/time/locale/de.kt:11]:
```suggestion
    companion object : GermanKlockLocale() {
```

File: klock/src/commonMain/kotlin/korlibs/time/locale/de.kt [hunk 2]
```diff
@@ -32,4 +78,46 @@ open class GermanKlockLocale : KlockLocale() {
 
 	override val formatTimeMedium = format("HH:mm:ss")
 	override val formatTimeShort = format("HH:mm")
+
+    enum class GermanOrdinal(val value: Int) {
```
Comment (soywiz) [inline klock/src/commonMain/kotlin/korlibs/time/locale/de.kt:82]: This enum should be removed

**Comment by soywiz** [comment]
Can add some tests covering these German locale ordinals?

**Comment by Bantolomeus** [comment]
sure

File: klock/src/commonMain/kotlin/korlibs/time/locale/de.kt (hunk 2 above)
Comment (Bantolomeus) [reply klock/src/commonMain/kotlin/korlibs/time/locale/de.kt:82]: My bad, wanted to delete it.

File: klock/src/commonMain/kotlin/korlibs/time/locale/de.kt (hunk 1 above)
Comment (Bantolomeus) [reply klock/src/commonMain/kotlin/korlibs/time/locale/de.kt:11]: done

**Comment by Bantolomeus** [comment]
I will add the test either later today or tomorrow.

**Comment by Bantolomeus** [comment]
I could not come up with a good test for the ordinals so I copied your english one. But that is kinda useless, since it does exactly the same as the german ordinal code since I can not add a suffix to the german ordinals. 
Please tell me if the test shall stay or if I should remove it.
And I changed the index 0 to "nullste" since you did it that way in english, to stay consistent.
If you have not further concerns you can merge the PR from my end.

**Comment by soywiz** [comment]
Looks good to me.
Don't remember the english version, but since it is not going to be accessed the first element could be an empty string or a null. Then it could be checked as `if (number !in 1..31) outofboundsexception`.
In any case, looks good to me as it is now 👍. Thanks for the PR @Bantolomeus !

**Comment by Bantolomeus** [comment]
Thanks for Guiding me through it. It was my pleasure and my very first open source contribution =D

**Comment by Bantolomeus** [comment]
The native windows Test failed, but it do not understand what the error is, something with coroutines, but nothing concrete. Was the Test flaky?

**Comment by soywiz** [comment]
It is flaky indeed. Nothing to do with this code. I have re-run it, and likely will be merged now.

**Pull request merged**